streams/src/test/java/org/apache/kafka/streams/processor/internals/StreamsProducerTest.java

```diff
@@ -66,27 +72,57 @@
         Collections.emptySet()
     );
 
-    private final ByteArraySerializer byteArraySerializer = new ByteArraySerializer();
-    private final Map<TopicPartition, OffsetAndMetadata> offsetsAndMetadata = mkMap(
-        mkEntry(new TopicPartition(topic, 0), new OffsetAndMetadata(0L, null))
+    private final StreamsConfig config = new StreamsConfig(mkMap(
```

Comment:
This is a meta comment for this test class: I think there are a lot of opportunities to reduce the 500+ LOC here sine many are duplicated code. For example, we can have a single mock KafkaProducer and have ALO / EOS_ALPHA / EOS_BETA streams producer wrapping the client supplier which would return the mock producer. And then in each unit test we could save creating the mock producer, creating the streams producer etc which are almost in every test function.

Reply:
We can do some of that but not all. If we use a `mock(Producer.class)` what you say works.

However, if we use `MockProducer` this does not work, because we need to setup the `MockProducer` either for non-eos or eos and thus we need two instances. Therefore, we also need an `StreamsProducer` for each case.

Reply:
Yes I'm talking about using a `mock(Producer.class)`.